packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -34,6 +34,61 @@ import {
 } from './shared/shared-hook-functions';
 import { toSaveSettings } from './to-save-settings';
 
+function mergeHookFunctions(...hookFunctions: IWorkflowExecuteHooks[]): IWorkflowExecuteHooks {
+	const result: IWorkflowExecuteHooks = {
+		nodeExecuteBefore: [],
+		nodeExecuteAfter: [],
+		workflowExecuteBefore: [],
+		workflowExecuteAfter: [],
+		sendResponse: [],
+		nodeFetchedData: [],
+	};
+	for (const hooks of hookFunctions) {
+		for (const key in hooks) {
+			if (hooks.hasOwnProperty(key)) {
+				result[key]!.push(...hooks[key]!);
+			}
+		}
+	}
+	return result;
+}
+
+function hookFunctionsWorkflowTelemetry(userId?: string): IWorkflowExecuteHooks {
+	const eventService = Container.get(EventService);
+	return {
+		workflowExecuteBefore: [
+			async function (this: WorkflowHooks): Promise<void> {
+				const { executionId, workflowData } = this;
+				eventService.emit('workflow-pre-execute', { executionId, data: workflowData });
```

Comment:
Odd that `workflow-pre-execute` doesn't need user ID but `workflow-post-execute` does.

Reply:
yeah, there are still a couple of inconsistencies. I'd like to change this in another PR though.